technical/20180530_hpm5g-radio-tests.md

```diff
@@ -0,0 +1,183 @@
+
+![Dragino HPM5G mPCIe radio module](../images/20180530_hpm5g-radio-tests.jpg?raw=true)
+![LibreRouter's 12 dBi antenna](../images/20180530_hpm5g-radio-tests2.jpg?raw=true)
+
+## Issues
+
+Using our ESPRESSObin setup, the HPM5G board was automatically identified and brought up on boot with the `ath9k` driver. Throughout testing, we observed a couple issues that we were able to work around.
+
+### Power Issues on the ESPRESSObin mPCIe
+
+During the initial setup and testing we noticed that the ESPRESSObin would restart at what seemed to be random times. We tried different combination of radio boards and ESPRESSObin boards but the issue was persistent. We tried to reach out to the ESPRESSObin community without any luck; however the Armbian community mentioned that this seemed like an issue with insufficient power. Identifying this as a potential power issue we tried using larger amperage power supplies (1.5 and 2 A at 12 V) however this did not seem to solve the problem.
+
+**Work Around**
+
+After identifying a potential power problem we tried to limit the amount of power the radio would need by reducing the TX power of the radio. We found that keeping it around 14 dBm would let the ESPRESSObin operate without reboot. Our initial testing was then done using a 14 dBm TX power.
+
+We did run into an issue where the TX power would not be persistent after being set, due to the Network Manager Service changing its value. The solution was to change the regulatory database to prevent TX power higher then 14 dBm. This held the TX power at 14 dBm.
+
+**Long-term Solution**
+
+This issue is specific to the ESPRESSObin, since the LibreRouter mPCIe interfaces are designed to meet the power requirements of the HPM5G at 19 dBm. In later parts of this document, we will also describe a board modification that will allow us to power the HPM5G properly on the ESPRESSObin.
+
+### Uninitialized MAC Addresses on the HPM5G
+
+Both HPM5G radios were brought up with the same MAC address of `00:02:03:04:05:06`, which prevented two devices from talking to each other.
+
+Additionally, the ESPRESSObin interfaces themselves seemed to be the same as well. This is an unrelated bug in Armbian.
+
+**Work Around**
+
+Setting the MAC address manually allowed the two devices to mesh together.
+
+The ESPRESSObin interface issue was resolved by regenerating a unique machine ID and rebooting:
+
+```
+sudo mv /etc/machine-id /etc/machine-id.old
+dbus-uuidgen | sudo tee /var/lib/dbus/machine-id
+sudo cp /var/lib/dbus/machine-id /etc/machine-id
+```
+
+**Long-term Solution**
+
+This issue is specific to the sample units and will not be an issue when the HPM5G is manufactured for production.
```

Comment:
Dragino posted "fix"  i have not yet tried it (instructions are for openwrt)
https://github.com/libremesh/librerouter/issues/20#issuecomment-392115592

Reply:
While the interface is down you can execute ONE of these two commands (Replacing wlan0 and the mac address as needed)

```
sudo ip link set dev wlan0 address 00:28:C7:0A:42:A2
sudo ifconfig wlan0 hw ether 00:28:C7:0A:42:A2
```

For survivability I added the line into the  mesh-point script just after bringing the interface down to set a new hard coded mac (different number for each device)